### PR TITLE
added user service, etc.

### PR DESCRIPTION
**Pull request by dvoet**
added createUser end point
remove requirement for first/last name
put resource endpoints under /api
implemented MockDirDao

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala [hunk 1]
```diff
@@ -93,16 +93,16 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
...
-          Seq("top", "inetorgperson").foreach(oc.add)
+          Seq("top", "inetOrgPerson").foreach(oc.add)
           myAttrs.put(oc)
 
           user.email.foreach { email =>
             myAttrs.put(new BasicAttribute(Attr.email, email.value))
           }
 
-          myAttrs.put(new BasicAttribute(Attr.sn, user.lastName))
-          myAttrs.put(new BasicAttribute(Attr.givenName, user.firstName))
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala:104]: if `givenName` is no longer used, can we remove it from the `Attr` object?

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  implicit val executionContext: ExecutionContext
+  val userService: UserService
+
+  def userRoutes: server.Route =
+    pathPrefix("user") {
+      requireUserInfo { userInfo =>
+        pathEndOrSingleSlash {
+          post {
+            complete {
+              userService.createUser(SamUser(userInfo.userId, Option(userInfo.userEmail))).map(user => StatusCodes.Created -> user)
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala:28]: i know it's not new in this PR, but does `userEmail` still need to be an `Option`?

**Review by MatthewBemis — APPROVED** [review]
1 minor comment + 1 minor question

otherwise :+1:

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala (hunk 1 above)
Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala:104]: I will address in another pr

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala (hunk 2 above)
Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala:28]: doesn't hurt... much

**Pull request merged**